### ISSUE TITLE
rainbow -- can't modify frozen string in recent ruby versions (bundler 1.14.2)

### ISSUE THREAD

**Issue by olbrich** [bundler/bundler#5352]
While installing 'rubocop' which depends on 'rainbow' in a gem that builds against older versions of ruby, I get the following error.

```
Gem::Ext::BuildError: ERROR: Failed to build gem native extension.

    can't modify frozen String

Gem files will remain installed in /Users/kolbrich/.rvm/gems/ruby-2.3.3/gems/rainbow-2.2.1 for inspection.
Results logged to /Users/kolbrich/.rvm/gems/ruby-2.3.3/extensions/x86_64-darwin-16/2.3.0/rainbow-2.2.1/gem_make.out
```

This works fine if the ruby version is 2.1.10 or 2.2.5, but fails for 2.3.3 and later.

## Versions

Bundler: 1.14.2
Ruby: 2.1.10, 2.2.5, 2.3.3, 2.4.0
Gem: 2.6.8
RVM: 1.26.10

## Gemfile

```
source 'https://rubygems.org'
ruby RUBY_VERSION
gemspec
```

## gemspec
```
# coding: utf-8
lib = File.expand_path('../lib', __FILE__)
$LOAD_PATH.unshift(lib) unless $LOAD_PATH.include?(lib)
require 'fhir_models/version'

Gem::Specification.new do |spec|
  spec.name          = 'fhir_models'
  spec.version       = FHIR::Models::VERSION
  spec.authors       = ['Jason Walonoski', 'Andre Quina', "Michael O'Keefe"]
  spec.email         = ['jwalonoski@mitre.org']

  spec.summary       = %q{A Gem for handling FHIR models in ruby}
  spec.description   = %q{A Gem for handling FHIR models in ruby}
  spec.homepage      = 'https://github.com/fhir-crucible/fhir_models'

  spec.files         = `git ls-files -z`.split("\x0").reject do |f|
    f.match(%r{^(test|spec|features)/})
  end
  spec.bindir        = 'exe'
  spec.executables   = spec.files.grep(%r{^exe/}) { |f| File.basename(f) }
  spec.require_paths = ['lib']

  spec.add_dependency 'nokogiri', '>= 1.6'
  spec.add_dependency 'date_time_precision', '>= 0.8'
  spec.add_dependency 'bcp47', '>= 0.3'
  spec.add_dependency 'mime-types', '>= 1.16', '< 3'

  spec.add_development_dependency 'bundler'
  spec.add_development_dependency 'rake'
  spec.add_development_dependency 'pry'
  spec.add_development_dependency 'test-unit'
  spec.add_development_dependency 'rspec'
  spec.add_development_dependency 'simplecov'
  spec.add_development_dependency 'nokogiri-diff'
  spec.add_development_dependency 'rubocop'
  spec.add_development_dependency 'codeclimate-test-reporter'
  spec.add_development_dependency 'guard-rspec'
  spec.add_development_dependency 'guard-test'
end
```

Failing Travis build: https://travis-ci.org/Mckesson-cds/fhir_models/jobs/194104180

Reverting to bundler 1.13.7 fixes this problem.

**Comment by olleolleolle** [comment]
@olbrich Latest rubygems 2.6.9 can be used to avoid this problem. `gem update --system`

**Comment by segiddins** [comment]
Yup, the root cause of this is a bug in ruby gems which is fixed in 2.6.9.

**Comment by olbrich** [comment]
👍 Updating rubygems fixes this issue, thanks!
